nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/FuzzyHashContent.java

```diff
@@ -0,0 +1,192 @@
+            "tlsh",
+            "tlsh",
+            "Uses TLSH (Trend 'Locality Sensitive Hash'). Note: FlowFile Content must be at least 512 characters long");
+
+    public static final PropertyDescriptor ATTRIBUTE_NAME = new PropertyDescriptor.Builder()
+            .name("ATTRIBUTE_NAME")
+            .displayName("Hash Attribute Name")
+            .description("The name of the FlowFile Attribute into which the Hash Value should be written. " +
+                    "If the value already exists, it will be overwritten")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("fuzzyhash.value")
+            .build();
+
+    public static final PropertyDescriptor HASH_ALGORITHM = new PropertyDescriptor.Builder()
+            .name("HASH_ALGORITHM")
+            .displayName("Hashing Algorithm")
+            .description("The hashing algorithm utilised")
+            .allowableValues(allowableValueSSDEEP, allowableValueTLSH)
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("Success")
+            .description("Any FlowFile that is successfully hashed will be sent to this Relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("Failure")
+            .description("Any FlowFile that is successfully hashed will be sent to this Relationship.")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(ATTRIBUTE_NAME);
+        descriptors.add(HASH_ALGORITHM);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        final String algorithm = context.getProperty(HASH_ALGORITHM).getValue();
```

Comment:
Also, maybe just get the PropertyDescriptor and use that for a comparison in your #read instead of comparing strings?  Doesn't seem like this is being used anywhere.